provisioner/ansible/provisioner.go

```diff
@@ -382,6 +391,21 @@ func validateFileConfig(name string, config string, req bool) error {
 	return nil
 }
 
+func validateDirectoryConfig(name string, config string, req bool) error {
```

Comment:
There's no need for `config` and `req`. This function is only called when `name` is guaranteed to be something other than the empty string.